state-machine_test.go

```diff
@@ -70,6 +70,7 @@ var (
 				PSKs:             &PSKMapCache{},
 				Certificates:     certificates,
 				RequireCookie:    true,
+				CookieHandler:    &defaultCookieHandler{},
```

Comment:
Is this needed? It seems like you do this by default. Wouldn't a better test be to show you could override the cookie handler.

Reply:
The `conn` sets the default value, but not the `ServerStateStart`. There are a bunch of tests I'd like here (at least: default cookie handler, custom cookie handler, and failed cookie validation), but the current format of the tests doesn't really allow to perform those.